config/main.py

```diff
@@ -542,32 +586,132 @@ def config():
 config.add_command(nat.nat)
 
 @config.command()
+@click.option('-n', '--namespace', help='Namespace name')
 @click.option('-y', '--yes', is_flag=True, callback=_abort_if_false,
-                expose_value=False, prompt='Existing file will be overwritten, continue?')
-@click.argument('filename', default='/etc/sonic/config_db.json', type=click.Path())
-def save(filename):
+                expose_value=False, prompt='Existing files will be overwritten, continue?')
+@click.argument('filename',nargs=-1)
+def save(namespace, filename):
     """Export current config DB to a file on disk."""
-    command = "{} -d --print-data > {}".format(SONIC_CFGGEN_PATH, filename)
-    run_command(command, display_cmd=True)
+    if namespace:
+        if not is_multi_asic():
+            click.echo("Namespace is not significant in a Single ASIC platform")
+            return None
+        if not validate_namespace(namespace):
+            click.echo("Invalid Namespace entered {}".format(namespace))
+            return None
+        if not len(filename):
+            inst = namespace[len(NAMESPACE_PREFIX)]
+            cfg_file = "/etc/sonic/config_db{}.json".format(inst)
+        else:
+            cfg_file = filename[0]
```

Comment:
I do not understand the logic here. what is filename[0]?

Reply:
Ok filename is returned as a tuple, as this is of type nargs=-1. But yes this is wrong in this case as here the tuple has only one element and I can just use cfg_file = filename.

Will fix it and update this and other similar places.